src/agora/txrx/workers/txrx_worker_argos.cc

```diff
@@ -0,0 +1,730 @@
+/**
+ * @file txrx_worker_argos.cc
+ * @brief Implementation of PacketTxRx datapath functions for communicating
+ * with real Argos hardware
+ */
+
+#include "txrx_worker_argos.h"
+
+#include <cassert>
+
+#include "logger.h"
+
+static constexpr bool kSymbolTimingEnabled = false;
+
+TxRxWorkerArgos::TxRxWorkerArgos(
+    size_t core_offset, size_t tid, size_t interface_count,
+    size_t interface_offset, Config* const config, size_t* rx_frame_start,
+    moodycamel::ConcurrentQueue<EventData>* event_notify_q,
+    moodycamel::ConcurrentQueue<EventData>* tx_pending_q,
+    moodycamel::ProducerToken& tx_producer,
+    moodycamel::ProducerToken& notify_producer,
+    std::vector<RxPacket>& rx_memory, std::byte* const tx_memory,
+    std::mutex& sync_mutex, std::condition_variable& sync_cond,
+    std::atomic<bool>& can_proceed, RadioConfig& radio_config)
+    : TxRxWorker(core_offset, tid, interface_count, interface_offset, config,
+                 rx_frame_start, event_notify_q, tx_pending_q, tx_producer,
+                 notify_producer, rx_memory, tx_memory, sync_mutex, sync_cond,
+                 can_proceed),
+      radio_config_(radio_config),
+      program_start_ticks_(0),
+      freq_ghz_(GetTime::MeasureRdtscFreq()) {}
+
+TxRxWorkerArgos::~TxRxWorkerArgos() = default;
+
+//Main Thread Execution loop
+void TxRxWorkerArgos::DoTxRx() {
+  PinToCoreWithOffset(ThreadType::kWorkerTXRX, core_offset_, tid_);
+
+  MLPD_INFO("TxRxWorkerArgos[%zu] has %zu:%zu total radios %zu\n", tid_,
+            interface_offset_, (interface_offset_ + num_interfaces_) - 1,
+            num_interfaces_);
+
+  running_ = true;
+  WaitSync();
+
+  if (num_interfaces_ == 0) {
+    MLPD_WARN("TxRxWorkerArgos[%zu] has no interfaces, exiting\n", tid_);
+    running_ = false;
+    return;
+  }
+
+  long long time0 = 0;
+  // When Hw Framer is enabled, frame time keeping is done by hardware.
+  // In particular, the Hw only forward useful receive (no Guard) symbols.
+  // In this mode, the format of rx timestamp (64 bits)is frame number at
+  // 32-bits msb and symbol number at 16-bit msb of 32-bit lsb.
+  // When Hw Framer is disabled (false), frame time keeping in done entirely
+  // in software as it is implemented here. All time domain symbols are transfered
+  // to the host by HW where software will decide how to handle each symbol.
+  if (Configuration()->HwFramer() == false) {
```

Comment:
I feel like we should make to separate function for hw_framer mode vs sw_framer, or even separate worker files? The file got too long after I added this.

Reply:
Will fix in next PR